pkg/apis/core/validation/validation.go

```diff
@@ -4421,6 +4424,24 @@ func validateSysctls(sysctls []core.Sysctl, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+var supportedSupplementalGroupsPolicy = sets.NewString(string(core.SupplementalGroupsPolicyMerge), string(core.SupplementalGroupsPolicyStrict))
+
+func validateSupplementalGroupsPolicy(supplementalGroupsPolicy *core.SupplementalGroupsPolicy, podOs *core.PodOS, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SupplementalGroupsPolicy) {
```

Comment:
This is not how we usually gate new fields.

The gate controls admission into the system.  Once it is in, we use it.  What that means practically is that you want to add this in k8s.io/kubernetes/pkg/api/pod `DropDisabledPodFields()` (see the other examples).

Then, in here, you do something like:

```
if gate enabled {
    if field is nil {
        field.Required(...)
    } else if field is not a valid value {
            field.Invalid(...)
    }
}
```

You also need to make sure that no updates are allowed when the gate is disabled.